go/worker/sentry/grpc/worker.go

```diff
@@ -152,6 +165,32 @@ func (g *Worker) worker() {
 	defer close(g.quitCh)
 	defer (g.cancelCtx)()
 
+	if g.upstreamConn == nil {
+		var numRetries uint
+
+		dialUpstream := func() error {
+			_, err := g.upstreamDialer(g.ctx)
+			if err != nil {
+				if numRetries < 60 {
```

Comment:
Could instead wrap the existing backoff with [`WithMaxRetries`](https://godoc.org/github.com/cenkalti/backoff#WithMaxRetries)